homeassistant/components/tuya/config_flow.py

```diff
@@ -379,6 +384,17 @@ def _get_climate_schema(curr_conf, device):
                     CONF_CURR_TEMP_DIVIDER,
                     default=curr_conf.get(CONF_CURR_TEMP_DIVIDER, 0),
                 ): vol.All(vol.Coerce(int), vol.Clamp(min=0)),
+                vol.Optional(
+                    CONF_PRECISION_OVERRIDE,
+                    default=curr_conf.get(CONF_PRECISION_OVERRIDE, CONF_DEFAULT),
+                ): vol.In(
+                    {
+                        CONF_DEFAULT: "Default",
+                        PRECISION_TENTHS: "Tenths",
+                        PRECISION_HALVES: "Halves",
+                        PRECISION_WHOLE: "Whole",
```

Comment:
You are mixing string and number here. I think you should define a `PRECISION_DEFAULT` constant as number instead of using `CONF_DEFAULT` or also the other values will be converted to string.

Reply:
Fixed